src/Middleware/OutputCaching/src/OutputCacheAttribute.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.OutputCaching;
+
+/// <summary>
+/// Specifies the parameters necessary for setting appropriate headers in output caching.
+/// </summary>
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+public class OutputCacheAttribute : Attribute, IPoliciesMetadata
+{
+    // A nullable-int cannot be used as an Attribute parameter.
+    // Hence this nullable-int is present to back the Duration property.
+    // The same goes for nullable-ResponseCacheLocation and nullable-bool.
+    private int? _duration;
+    private bool? _noStore;
+
+    private List<IOutputCachingPolicy>? _policies;
+
+    /// <summary>
+    /// Gets or sets the duration in seconds for which the response is cached.
+    /// </summary>
+    public int Duration
```

Comment:
Internally it accepts a `Timestamp`, should we expose it here too in case users want to cache things with shorter times, or assume that custom profiles are sufficient for that?